### PR TITLE
raise ValueError if strings have size 0. Add test

### LINKED ISSUE #210: make sure to error on zero sized column types (e.g., 'U0', 'S0')

### PR DESCRIPTION
**Pull request by esheldon**
closes #210

### CODE REVIEW

File: fitsio/test.py [hunk 1]
```diff
@@ -1316,6 +1316,10 @@ def testTableWriteBadString(self):
 
                     except ValueError:
                         got_error=True
+                    except TypeError:
```
Comment (beckermr) [inline fitsio/test.py]: Can we scope this error to only coming from numpy?

Comment (esheldon) [reply fitsio/test.py]: It doesn't look to be subclassed.  do you have something else in mind?

Comment (beckermr) [reply fitsio/test.py]: Maybe test against the exact error string from numpy?

Comment (beckermr) [reply fitsio/test.py]: Or a substring of it?

```python
except TypeError as e:
    if str(e) == 'Empty data-type':
         got_error = True
    else:
        raise
```

Comment (beckermr) [reply fitsio/test.py]: Not sure if raise is right for the else clause

**Pull request merged**